src/workerd/io/io-context.c++

```diff
@@ -1203,8 +1203,11 @@ void IoContext::runImpl(Runnable& runnable, bool takePendingEvent,
       // Check if we hit a limit.
       limitEnforcer->requireLimitsNotExceeded();
 
-      // That should have thrown, so we shouldn't get here.
-      KJ_FAIL_ASSERT("script terminated for unknown reasons");
+      // That should have thrown, so we shouldn't get here. If we do end up here tho,
+      // it is because of some other kind of terminal error thrown by v8. An OOM,
+      // perhaps? Either way there's not much else we can do here but fail. But
+      // try try not to fail too noisily, shall we?
+      KJ_FAIL_ASSERT("NOSENTRY script terminated for unknown reasons");
```

Comment:
So usually we leave these sort of fail asserts so that we get a consistent sentry event that we can look up in kibana. Are we not logging anything meaningful when we assert? Is this sort of failure inactionable? Do we actually want worker_do_not_log instead?

Reply:
On this error there's really nothing meaningfully logged... just essentially "an unknown error happened here". Sure, it's likely being triggered by something somewhere but it's not at all clear *where* or *what*... most likely it's an OOM kill or interrupt happening that's causing a terminal error in the isolate but by this point in time the log message doesn't give us enough detail to really do anything about it and the stack that is reported isn't helpful either.

To @irvinebroque's point, exposing a counter for this would likely be more useful than the sentry log.